packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -969,55 +963,52 @@ All @auth directives used on field definitions are performed when the field is r
+        if (names) {
+            names.forEach( (name) => {
                 this.addSubscriptionResolvers(ctx, rules, parent, level, name)
             })
-        } else {
-            this.addSubscriptionResolvers(ctx, rules, parent,
-                level, graphqlName(ON_CREATE_FIELD + toUpper(parent.name.value)))
         }
     }
 
     // OnUpdate Subscription
     private protectOnUpdateSubscription(ctx: TransformerContext, rules: AuthRule[],
-        parent: ObjectTypeDefinitionNode, level: string, onUpdate?: string[]) {
-        if (onUpdate) {
-            onUpdate.forEach( (name) => {
+        parent: ObjectTypeDefinitionNode, modelConfiguration: ModelDirectiveConfiguration) {
+        const names = modelConfiguration.getNames('onUpdate');
+        const level = modelConfiguration.getName('level') as ModelSubscriptionLevel;
+        if (names) {
+            names.forEach( (name) => {
                 this.addSubscriptionResolvers(ctx, rules, parent, level, name)
             })
-        } else {
-            this.addSubscriptionResolvers(ctx, rules, parent,
-                level, graphqlName(ON_UPDATE_FIELD + toUpper(parent.name.value)))
         }
     }
 
     // OnDelete Subscription
     private protectOnDeleteSubscription(ctx: TransformerContext, rules: AuthRule[],
-        parent: ObjectTypeDefinitionNode, level: string, onDelete?: string[]) {
-        if (onDelete) {
-            onDelete.forEach( (name) => {
+        parent: ObjectTypeDefinitionNode, modelConfiguration: ModelDirectiveConfiguration) {
+        const names = modelConfiguration.getNames('onDelete');
+        const level = modelConfiguration.getName('level') as ModelSubscriptionLevel;
+        if (names) {
+            names.forEach( (name) => {
                 this.addSubscriptionResolvers(ctx, rules, parent, level, name)
             })
-        } else {
-            this.addSubscriptionResolvers(ctx, rules, parent,
-                level, graphqlName(ON_DELETE_FIELD + toUpper(parent.name.value)))
         }
     }
 
     // adds subscription resolvers (request / response) based on the operation provided
     private addSubscriptionResolvers(ctx: TransformerContext, rules: AuthRule[],
-        parent: ObjectTypeDefinitionNode, level: string, fieldName: string) {
+        parent: ObjectTypeDefinitionNode, level: ModelSubscriptionLevel, fieldName: string) {
         const resolverResourceId = ResolverResourceIDs.ResolverResourceID("Subscription", fieldName);
         const resolver = this.resources.generateSubscriptionResolver(fieldName);
-        // If the data source does not exist it is created and added as a resource for PUBLIC && ON levels
+        // If the data source does not exist it is created and added as a resource for public && on levels
         const noneDS = ctx.getResource(ResourceConstants.RESOURCES.NoneDataSource)
 
         // add the rules in the subscription resolver
         if (!rules || rules.length === 0) {
```

Comment:
optimization: This should be an early return from the function, since if there are no rules we don't need to generate anything, do we?